Vector comparison impr

pointhi opened the pull request:
Implement various operators for integer vectors and normal integer variables of type ```i1```, ```i8```, ```i16``` and ```i64```:

grimmerm commented on projects/com.oracle.truffle.llvm.nodes/src/com/oracle/truffle/llvm/nodes/op/arith/integer/LLVMI1ArithmeticNode.java:
```diff
@@ -44,4 +45,67 @@ protected boolean executeI1(boolean left, boolean right) {
+        @Specialization
+        protected boolean div(boolean left, boolean right) {
+            if (!right) {
+                CompilerAsserts.neverPartOfCompilation();
```
That should be CompilerDirectives.transferToInterpreter

grimmerm commented on projects/com.oracle.truffle.llvm.nodes/src/com/oracle/truffle/llvm/nodes/op/arith/integer/LLVMI1ArithmeticNode.java:
```diff
@@ -44,4 +45,67 @@ protected boolean executeI1(boolean left, boolean right) {
+        @Specialization
+        protected boolean udiv(boolean left, boolean right) {
+            if (!right) {
+                CompilerAsserts.neverPartOfCompilation();
```
same here

grimmerm commented on projects/com.oracle.truffle.llvm.nodes/src/com/oracle/truffle/llvm/nodes/op/arith/vector/LLVMI1VectorArithmeticNode.java:
```diff
@@ -38,10 +38,59 @@
 public abstract class LLVMI1VectorArithmeticNode extends LLVMExpressionNode {
 
-    public abstract static class LLVMI1VectorAddNode extends LLVMI1VectorArithmeticNode {
+    public abstract static class LLVMI1VectorAddNode extends LLVMI8VectorArithmeticNode {
```
That does not look right.
how can I1 extend I8?

grimmerm commented on projects/com.oracle.truffle.llvm.runtime/src/com/oracle/truffle/llvm/runtime/vector/LLVMI1Vector.java:
```diff
@@ -88,6 +89,85 @@ private static LLVMI1Vector doOperation(LLVMI1Vector lhs, LLVMI1Vector rhs, Oper
+            @Override
+            public boolean eval(boolean a, boolean b) {
+                if (!b) {
+                    CompilerAsserts.neverPartOfCompilation();
```
transferToInterpreter

pointhi commented:
Thanks @grimmerm for outlining some mistakes on my side.

I fixed them in https://github.com/pointhi/sulong/commit/6ad153c3d9f611392085687783fa3cdeb91e152d, and squashed all commits together
